File: frontend/src/jmix/entities/Atmosphere.ts
import { AtmosphericGas } from "./AtmosphericGas";
export class Atmosphere {
  static NAME = "Atmosphere";
  id?: string;
  description?: string | null;
  pressure?: any | null;
  gases?: AtmosphericGas[] | null;
}
export type AtmosphereViewName = "_base" | "_instance_name" | "_local";
export type AtmosphereView<V extends AtmosphereViewName> = V extends "_base"
  ? Pick<Atmosphere, "id" | "description" | "pressure">
  : V extends "_instance_name"
  ? Pick<Atmosphere, "id" | "description">
  : V extends "_local"
  ? Pick<Atmosphere, "id" | "description" | "pressure">
  : never;

File: src/main/java/com/company/spacetrans/entity/Atmosphere.java
package com.company.spacetrans.entity;

import io.jmix.core.entity.annotation.JmixGeneratedValue;
import io.jmix.core.metamodel.annotation.Composition;
import io.jmix.core.metamodel.annotation.InstanceName;
import io.jmix.core.metamodel.annotation.JmixEntity;

import javax.persistence.*;
import java.util.List;
import java.util.UUID;

@JmixEntity
@Table(name = "ATMOSPHERE")
@Entity
public class Atmosphere {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private UUID id;

    @InstanceName
    @Column(name = "DESCRIPTION")
    private String description;

    @Column(name = "PRESSURE")
    private Double pressure;

    @Composition
    @OneToMany(mappedBy = "atmosphere")
    private List<AtmosphericGas> gases;

    public List<AtmosphericGas> getGases() {
        return gases;
    }

    public void setGases(List<AtmosphericGas> gases) {
        this.gases = gases;
    }

    public Double getPressure() {
        return pressure;
    }

    public void setPressure(Double pressure) {
        this.pressure = pressure;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }
}